src/components/Widget/index.js

```diff
@@ -30,17 +35,83 @@ class Widget extends Component {
   }
 
   componentDidMount() {
-    const { socket } = this.props;
+    const { socket, storage } = this.props;
 
     socket.on('bot_uttered', (botUttered) => {
       this.messages.push(botUttered);
     });
 
+    const local_id = this.getSessionId();
+
+    /* Request a session from server */
```

Comment:
Looks like session is a local browser object not a request from the server

Reply:
Moves the line down and changed comments